yolo_to_yolo/data_transformers.py

```diff
@@ -47,3 +47,137 @@ def __call__(self, input_data: YoloImageData) -> Iterable[YoloImageData]:
     def include(classname: str) -> bool:
         return classname != 'background' and len(classname) > 1
 
+class BBoxToCropTransformer(YoloDataTransformer):
+    """
+    Converts bounding box labels to cropped images
+
+    Args:
+    min_size: Minimum size of the cropped image (width, height)
+    min_padding: Padding to add around the bounding box
+    min_char_overlap: Minimum overlap between shape and character bounding boxes to consider them a match (0 to 1)
+        (This is used to filter out char bboxes that don't overlap enough with the shape bboxes. Default is 0, meaning it just has to overlap a little bit to be considered a match)
+    """
+    def __init__(self, min_size: tuple[int, int] = (0, 0), min_padding: int = 0, min_char_overlap: float = 0):
+        self.min_size = min_size
+        self.min_padding = min_padding
+        self.min_char_overlap = min_char_overlap
+
+    def __call__(self, input_data: YoloImageData) -> Iterable[YoloImageData]:
+        img_height, img_width = input_data.image.shape[:2]
+        
+        # I hope this is right
+        shape_labels = [label for label in input_data.labels if "shape" in label.classname]
+        char_labels = [label for label in input_data.labels if "character" in label.classname]
+        
+        for shape_label in shape_labels:
+            if isinstance(shape_label.location, YoloBbox):
+                shape_bbox = shape_label.location
+                best_char_label = None
+                best_overlap = self.min_char_overlap
+
+                # Find best match for character bbox (basically the one that overlaps the most with the shape bbox)
+                for char_label in char_labels:
+                    if isinstance(char_label.location, YoloBbox):
+                        char_bbox = char_label.location
+                        overlap = self._calculate_iou(shape_bbox, char_bbox)
+
+                        if overlap > best_overlap:
+                            best_overlap = overlap
+                            best_char_label = char_label
+
+                x1, y1, x2, y2 = self._get_combined_bbox(shape_bbox, best_char_label, img_width, img_height)
```

Comment:
There's a type error on `best_char_label` because it can be None. Just print a warning and `continue` if you can't find one instead of making it an error condition.